WordPress/gutenberg
# `ColorPalette`: refine custom color button's label

## What?


Change the label for the custom color button in the `ColorPalette` component:

- format the color value to always be upper case, and hide the starting `#` character
- display the color name

<details>

<summary>See more details</summary>

For reference, this is how this button looks like in `trunk`

![image (1)](https://user-images.githubusercontent.com/1083581/157872196-f4909cc3-8309-4407-abd2-4a175294b9ed.png)

And this is how it _should_ look like:

![image](https://user-images.githubusercontent.com/1083581/157872416-0a80dc50-d8ac-4c8f-875f-3a2ac7e30595.png)
</details>

## Why?


Iterating and refining our UIs

## How?


1. Added a utility function that is able to find the color name, given its hex value, by matching it against the color palette(s)
2. Used the `Flex` and `Truncate` components (and a couple of custom styles) to implement the layout changes
3. Added a fix to `fieldset` to make sure that it does't grow larger than its container.

## Testing Instructions





- Check the component in Storybook / Global Styles (in the theme editor) / Post editor (e.g. by setting a background color for a group block)
- Focus your attention on the custom color button, in the `ColorPalette` component:
    - The color value should be uppercase and stripped from the leading `#` character
    - The color name should be displayed. If the current color can't be matched against the current paletted of predefined named colors, the text `Custom` should be displayed instead
    - Make sure that, in case of a color with a very long name, the text is truncated correctly.

## Screenshots or screencast 

**In Storybook:**

https://user-images.githubusercontent.com/1083581/157878761-41d43388-b72b-4ed1-9921-95815e1e3c77.mp4

**In the theme editor:**

https://user-images.githubusercontent.com/1083581/157874005-6f997ffc-43d3-4b0c-894e-12114aa88a6f.mp4

**In the post editor:**

https://user-images.githubusercontent.com/1083581/157875711-94cb9db8-7f80-4f8e-99ce-0b9d2d34cd60.mp4
